### ISSUE TITLE
VRT_VIRTUAL_OVERVIEWS=YES has no effect

### ISSUE THREAD

**Issue by craigds** [OSGeo/gdal#8962]
## Expected behavior and actual behavior.

Expected: When creating/updating a warped VRT, setting `VRT_VIRTUAL_OVERVIEWS=YES` should cause gdalwarp or gdaladdo to add an `OverviewList` element.

Actual: no `OverviewList` is created.

## Steps to reproduce the problem.

* start with a TIF file with overviews
* warp to VRT:
```console
rm -f warped.vrt
VRT_VIRTUAL_OVERVIEWS=YES gdalwarp -t_srs EPSG:3857 auckland-0075m-urban-aerial-photos-2017.tif warped.vrt
```

* observe that `warped.vrt` contains no `OverviewList` element
* Try `gdaladdo`:
```console
VRT_VIRTUAL_OVERVIEWS=YES gdaladdo warped.vrt 2 4 8
```

* Still no `OverviewList` element

I'll note that `gdalbuildvrt mosaic.vrt *.tif` _does_ produce the `OverviewList` (even without the config option), but I'm after a way to add it to a warped VRT.

## Operating system

MacOS 14.1.1

## GDAL version and provenance

GDAL 3.7.2

**Comment by craigds** [comment]
For completeness I also tried Python:

```python
>>> from osgeo import gdal
>>> gdal.UseExceptions()
>>> gdal.SetConfigOption('VRT_VIRTUAL_OVERVIEWS', 'YES')
>>> ds = gdal.OpenEx('warped.vrt')
>>> ds.BuildOverviews(resampling='NEAREST', overviewlist=[2,4,8])
0
```

This didn't work either; still no `OverviewList`

**Comment by rouault** [comment]
> This didn't work either; still no `OverviewList`

yes, VRT_VIRTUAL_OVERVIEWS is not supported for warped VRTs. Note that adding explicitly a ``<OverviewList>`` element in the XML is possible, but not recommended, as the warped overviews will use the full resolution source dataset, so you won't get much speed-up.
If the source dataset of the warped VRT has itself overviews, then the warped VRT will implicitly create the appropriate overview levels.
```
$ gdal_translate byte.tif in.tif -outsize 1024 1024
$ gdaladdo in.tif 2 4
$ gdalwarp in.tif warped.vrt
$ gdalinfo warped.vrt
[...]
  Overviews: 512x512, 256x256
```
So basically there's nothing to do

**Comment by craigds** [comment]
> Note that adding explicitly a <OverviewList> element in the XML is possible, but not recommended, as the warped overviews will use the full resolution source dataset, so you won't get much speed-up.

In my testing I did achieve a very clear 8x speedup by adding `OverviewList` with `2 4 8`.

**Comment by craigds** [comment]
>  If the source dataset of the warped VRT has itself overviews, then the warped VRT will implicitly create the appropriate overview levels.

perhaps my exact more-complicated problem is causing this to not work effectively.

The actual problem I'm trying to solve is a downsampled warped mosaic of thousands of tiles.

So what I'm doing is:

* create a mosaic VRT of all the source tiles using `gdalbuildvrt`. (note: this has a OverviewList element automatically)
* create a warped VRT from the mosaic VRT (this doesn't get an OverviewList)
* `gdal_translate` from the warped VRT with a (much) lower resolution

If I don't add the `OverviewList` to the warped VRT, it is very slow and does 8x the IO. When I add the OverviewList it is a very clear speedup.

**Comment by rouault** [comment]
> If I don't add the `OverviewList` to the warped VRT, it is very slow and does 8x the IO

weird, works for me:
```
$ gdal_translate byte.tif in.tif -outsize 1024 1024
$ gdal_translate in.tif left.tif -srcwin 0 0 512 1024
$ gdal_translate in.tif right.tif -srcwin 512 0 512 1024
$ gdalbuildvrt mosaic.vrt left.tif right.tif
$ VRT_VIRTUAL_OVERVIEWS=YES gdaladdo mosaic.vrt 2 4 8
$ gdalwarp mosaic.vrt warped.vrt -overwrite   # it doesn't have an explicit OverviewList, but will create implicitly ones matching the ones of mosaic.vrt
$ gdalinfo warped.vrt | grep Overview
  Overviews: 512x512, 256x256, 128x128
```

**Comment by craigds** [comment]
Interesting. Thanks for having a look. I'll do some more testing and see if I can clarify.

**Comment by rouault** [comment]
> I'll do some more testing and see if I can clarify.

does your mosaic.vrt report overviews (check with gdalinfo) ?

**Comment by craigds** [comment]
yes, the mosaic reports overviews.

(still debugging the rest)

**Comment by craigds** [comment]
you're right, there's no clear bug in GDAL. Now that you've pointed it out, it actually works fine if I use gdal_translate on the CLI or via the Python bindings to generate a stripped image.

What I'm _actually_ actually doing is using Mapnik (!) to render a map of a certain size into a PDF file. It uses `gdal.RasterIO` - and I think the code for that call is [here](https://github.com/koordinates/mapnik/blob/1dca770da7003583ad471c3a3d073b1d8b967eaa/plugins/input/gdal/gdal_featureset.cpp#L610-L623)

I have just tried the equivalent call from Python which looks like:

```python
ds = gdal.OpenEx('warped.vrt')
data = ds.ReadRaster(buf_xsize=17688, buf_ysize=18368, buf_type=gdal.GDT_Byte)
```

... but this doesn't trigger the issue; it appears to use the source overviews just fine.

So I'm not sure what is going on inside mapnik to prevent this from working. It's likely to be a mapnik issue or an environmental issue rather than a GDAL bug, so I'll close this ticket.

Thanks for the assistance :)
